services/src/main/java/org/keycloak/crypto/ServerECDSASignatureSignerContext.java

```diff
@@ -0,0 +1,66 @@
+package org.keycloak.crypto;
+
+import org.bouncycastle.asn1.ASN1InputStream;
+import org.bouncycastle.asn1.ASN1Integer;
+import org.bouncycastle.asn1.ASN1Primitive;
+import org.bouncycastle.asn1.ASN1Sequence;
+import org.bouncycastle.asn1.x9.X9IntegerConverter;
+import org.keycloak.models.KeycloakSession;
+
+import java.io.IOException;
+import java.security.PrivateKey;
+import java.security.Signature;
+
+public class ServerECDSASignatureSignerContext extends AsymmetricSignatureSignerContext {
+
+    public ServerECDSASignatureSignerContext(KeycloakSession session, String algorithm) throws SignatureException {
+        super(getKey(session, algorithm));
+    }
+
+    public ServerECDSASignatureSignerContext(KeyWrapper key) {
+        super(key);
+    }
+
+    private static KeyWrapper getKey(KeycloakSession session, String algorithm) {
```

Comment:
Perhaps `ServerAsymmetricSignatureSignerContext.getKey` could be made package-protected and called instead of introducing this new method.